dll/win32/shell32/wine/shell32_main.c

```diff
@@ -541,55 +570,20 @@ DWORD_PTR WINAPI SHGetFileInfoW(LPCWSTR path,DWORD dwFileAttributes,
     /* get the displayname */
     if (SUCCEEDED(hr) && (flags & SHGFI_DISPLAYNAME))
     {
-        if (flags & SHGFI_USEFILEATTRIBUTES && !(flags & SHGFI_PIDL))
-        {
-            lstrcpyW (psfi->szDisplayName, PathFindFileNameW(szFullPath));
-        }
-        else if (psfParent)
-        {
-            STRRET str;
-            hr = IShellFolder_GetDisplayNameOf( psfParent, pidlLast,
-                                                SHGDN_INFOLDER, &str);
-            StrRetToStrNW (psfi->szDisplayName, MAX_PATH, &str, pidlLast);
-        }
+        STRRET str;
+        psfi->szDisplayName[0] = UNICODE_NULL;
+        hr = IShellFolder_GetDisplayNameOf(psfParent, pidlLast, SHGDN_INFOLDER, &str);
+        if (SUCCEEDED(hr))
+            StrRetToStrNW(psfi->szDisplayName, MAX_PATH, &str, pidlLast);
     }
 
     /* get the type name */
     if (SUCCEEDED(hr) && (flags & SHGFI_TYPENAME))
     {
-        if (!(flags & SHGFI_USEFILEATTRIBUTES) || (flags & SHGFI_PIDL))
-        {
-            _ILGetFileType(pidlLast, psfi->szTypeName, _countof(psfi->szTypeName));
-        }
-        else
-        {
-            if (dwFileAttributes & FILE_ATTRIBUTE_DIRECTORY)
-                strcatW (psfi->szTypeName, L"Folder");
-            else 
-            {
-                WCHAR sTemp[64];
-
-                lstrcpyW(sTemp,PathFindExtensionW(szFullPath));
-                if (sTemp[0] == 0 || (sTemp[0] == '.' && sTemp[1] == 0))
-                {
-                    /* "name" or "name." => "File" */
-                    lstrcpynW (psfi->szTypeName, L"File", 64);
-                }
-                else if (!( HCR_MapTypeToValueW(sTemp, sTemp, 64, TRUE) &&
-                    HCR_MapTypeToValueW(sTemp, psfi->szTypeName, 80, FALSE )))
-                {
-                    if (sTemp[0])
-                    {
-                        lstrcpynW (psfi->szTypeName, sTemp, 64);
-                        strcatW (psfi->szTypeName, L" file");
-                    }
-                    else
-                    {
-                        lstrcpynW (psfi->szTypeName, L"File", 64);
-                    }
-                }
-            }
-        }
+        /* FIXME: Use IShellFolder2::GetDetailsEx */
+        UINT col = _ILIsDrive(pidlLast) ? 1 : 2; /* SHFSF_COL_TYPE */
```

Comment:
It's not possible to use the define directly?

Reply:
It is not available to the wine part, no. This will go away when GetDetailsEx gets implemented (soon-ish I hope).